### PR TITLE
Added fixer for newline-before-return rule

### PR DESCRIPTION
**Pull request by jukben**
#### PR checklist

- [ ] Addresses an existing issue: #0000
- [x] New feature, bugfix, or enhancement
  - [x] Includes tests
- [ ] Documentation update

#### Overview of change:
I've realized that **newline-before-return** didn't have a fixer which seems to be quite straightforward. Here are my two cents. 

I've added tests and everything seems OK.

#### CHANGELOG.md entry:

[new-fixer] added fixer for **newline-before-return** rule

### CODE REVIEW

**Comment by palantirtech** [comment]
Thanks for your interest in [palantir/tslint](https://github.com/palantir/tslint), @jukben! Before we can accept your pull request, you need to sign our contributor license agreement - just visit https://cla.palantir.com/ and follow the instructions. Once you sign, I'll automatically update this pull request.

File: src/rules/newlineBeforeReturnRule.ts [hunk 1]
```diff
@@ -77,10 +84,18 @@ class NewlineBeforeReturnWalker extends Lint.AbstractWalker<void> {
...
         const prevLine = ts.getLineAndCharacterOfPosition(this.sourceFile, prev.end).line;
-
         if (prevLine >= line - 1) {
+            const indentationCurrent = this.getIndentation(node);
+            const indentationPrev = this.getIndentation(prev);
+
+            const fixer = Lint.Replacement.replaceFromTo(
+                start - indentationCurrent.length,
+                start,
+                line === prevLine ? `\n\n${indentationPrev}` : `\n${indentationCurrent}`,
```
Comment (JoshuaKGoldberg) [inline src/rules/newlineBeforeReturnRule.ts]: > `\n`

This'll need to adjust to projects where `\r\n` is the line feed, such as Git with CRLF (aka Windows). `curlyRule.ts` checks the last character of a relevant source line to see which to use, for example.

File: src/rules/newlineBeforeReturnRule.ts [hunk 2]
```diff
@@ -53,6 +54,12 @@ class NewlineBeforeReturnWalker extends Lint.AbstractWalker<void> {
         return ts.forEachChild(sourceFile, cb);
     }
 
+    private getIndentation(node: ts.Node): string {
```
Comment (JoshuaKGoldberg) [inline src/rules/newlineBeforeReturnRule.ts]: Seems like this is shared logic between this file, `indentRule.ts`, and `curlyRule.ts`. How about combining their logic into a utils function or two (`src/utils.ts` or similar)? Does that work?

**Comment by adidahiya** [comment]
@jukben have you signed the CLA? also please avoid force-pushing to PRs in this project, it's easier to review if you just add additional commits. they all get squashed at the end of the PR.

**Comment by jukben** [comment]
@adidahiya yes I have signed the CLA already. 

> You signed the individual CLA on January 26, 2019 at 4:19 PM.

Anyway, sorry for the force-push I’m aware of this, I haven’t been adding new commits I was just trying to unblock/resolve this because of conflicts comming from master. Should I use merge strategy next time?

**Comment by jukben** [comment]
@adidahiya Makes sense, next time I'll do it like this to avoid any force-push when PR is being reviewed. 👍 

screenshot 2019-02-03 at 17 29 48

Is this enough? 🙏

**Comment by jukben** [comment]
@adidahiya :shipit: ❤️

Could I somewhere get the information how releases are planned? Just to plan when it will be possible to intercorporate it _cleanly_ to my project?

**Comment by adidahiya** [comment]
the release schedule is pretty ad-hoc right now, but I'll try to cut a release this week

**Pull request merged**